### PR TITLE
Add pgBouncer in front of postgresql

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- [x] add [pgBouncer:1.15.0](https://hub.docker.com/r/pgbouncer/pgbouncer) service in simcore docker-compose
- [x] separate catching of exceptions in webserver's director-v2 client
- [x] increase number of clients accessing pgBouncer in all services to default to min 10 -> 100
- [x] update adminer to 4.8.0, 4.7.x was not happy with pgbouncer (⚠️ devops this needs to be updated)
- [x] adminer can access directly postgres through the postgres_network, or pgbouncer (default) through the simcore_default network (⚠️ devops this needs to be updated)
- [x] .env-devel POSTGRES_HOST and POSTGRES_ENDPOINT changed to ```pgbouncer``` (⚠️ devops this needs to be updated)
- [x] unit/integration/system tests adapted. now when using postgres in a test, ensure pgbouncer is also there (a warning message was added in the postgres_service.py)
- [x] removed connect timeout in aiohttp clients (this timeout is for acquiring a connection in the pool, not for peer connection)
- [x] added a max_concurrency parameter to logged_gather so gathering calls can be limited

pgBouncer is handling a pool of threads in front of the postgreSQL database. It can handle 1000s of connections while keeping only 100 to any number of databases behind it, thus it should alleviate issues with the "too many clients" error message in postgres.

```
client1**\
client2** - ---connect  10-100 times to pgBouncer  which connects up to 100 times to postgresql
...         /
clientN /
```

### CODE REVIEW

File: packages/postgres-database/tests/docker-compose.yml [hunk 1]
```diff
@@ -1,7 +1,27 @@
...
+    image: pgbouncer/pgbouncer:1.15.0
+    init: true
+    environment:
+      - DATABASES_HOST=postgres
+      - DATABASES_PORT=5432
+      - DATABASES_USER=test
+      - DATABASES_PASSWORD=test
+      - DATABASES_DBNAME=test
+      - PGBOUNCER_LISTEN_PORT=5432
+      - PGBOUNCER_LISTEN_ADDE=*
```
Comment (mguidon) [inline packages/postgres-database/tests/docker-compose.yml:13]: ADDRESS?

**Review by mguidon — APPROVED** [review]
Looking forward to see this in action!

**Review by pcrespov — APPROVED** [review]
Pair reviewed! Great!

File: packages/postgres-database/tests/docker-compose.yml (hunk 1 above)
Comment (sanderegg) [reply packages/postgres-database/tests/docker-compose.yml:13]: no it's a stupid env name ;)

**Review by GitHK — COMMENTED** [review]
nice 👍

**Pull request merged**